File: mock-facility-management/src/main/java/de/vkb/dojo/es/facilityManagement/model/ref/FeedbackReference.java
package de.vkb.dojo.es.facilityManagement.model.ref;

import com.fasterxml.jackson.annotation.*;

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeName("feedback")
public class FeedbackReference implements Reference {
    private final String id;

    @JsonCreator
    public FeedbackReference(
            @JsonProperty("id") String id
    ) {
        this.id = id;
    }

    @JsonGetter
    public String getId() {
        return id;
    }

    @Override
    public String getPath() {
        return "/feedback/" + id;
    }
}

File: todo-reservation-micronaut/src/main/kotlin/de/vkb/dojo/es/common/model/ref/FeedbackReference.kt
package de.vkb.dojo.es.common.model.ref

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonTypeName

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeName("feedback")
data class FeedbackReference(
    override val id: String
): Reference {
    override val path: String = "/feedback/${id}"
}
